utilities/test_suite/HOST/Tensor_host_audio.cpp

```diff
@@ -0,0 +1,307 @@
+    int numRuns = atoi(argv[5]);
+    int batchSize = atoi(argv[6]);
+    char *dst = argv[7];
+
+    // Set case names
+    string funcName = audioAugmentationMap[testCase];
+    if (funcName.empty())
+    {
+        if (testType == 0)
+            printf("\ncase %d is not supported\n", testCase);
+
+        return -1;
+    }
+
+    // Initialize tensor descriptors
+    RpptDesc srcDesc, dstDesc;
+    RpptDescPtr srcDescPtr, dstDescPtr;
+    srcDescPtr = &srcDesc;
+    dstDescPtr = &dstDesc;
+
+    // Set src/dst data types in tensor descriptors
+    if (inputBitDepth == 2)
+    {
+        srcDescPtr->dataType = RpptDataType::F32;
+        dstDescPtr->dataType = RpptDataType::F32;
+    }
+
+    // Other initializations
+    int missingFuncFlag = 0;
+    int i = 0, j = 0, fileCnt = 0;
+    int maxChannels = 0;
+    int maxSrcWidth = 0, maxSrcHeight = 0;
+    int maxDstWidth = 0, maxDstHeight = 0;
+    unsigned long long iBufferSize = 0;
+    unsigned long long oBufferSize = 0;
+    static int noOfAudioFiles = 0;
+
+    // String ops on function name
+    char src1[1000];
+    strcpy(src1, src);
+    strcat(src1, "/");
+    string func = funcName;
+
+    // Get number of audio files
+    vector<string> audioNames, audioFilePath;
+    search_files_recursive(src, audioNames, audioFilePath, ".wav");
+    noOfAudioFiles = audioNames.size();
+    if (noOfAudioFiles < batchSize || ((noOfAudioFiles % batchSize) != 0))
+    {
+        replicate_last_file_to_fill_batch(audioFilePath[noOfAudioFiles - 1], audioFilePath, audioNames, audioNames[noOfAudioFiles - 1], noOfAudioFiles, batchSize);
+        noOfAudioFiles = audioNames.size();
+    }
+
+    // Initialize the AudioPatch for source
+    Rpp32s *srcLengthTensor = (Rpp32s *) calloc(batchSize, sizeof(Rpp32s));
+    Rpp32s *channelsTensor = (Rpp32s *) calloc(batchSize, sizeof(Rpp32s));
+    RpptImagePatch *srcDims = (RpptImagePatch *) calloc(batchSize, sizeof(RpptImagePatch));
+    RpptImagePatch *dstDims = (RpptImagePatch *) calloc(batchSize, sizeof(RpptImagePatch));
+
+    // Find max audio dimensions in the input dataset
+    maxSrcHeight = 1;
+    maxDstHeight = 1;
+    for (int cnt = 0; cnt < noOfAudioFiles ; cnt++)
+    {
+        SNDFILE	*infile;
+        SF_INFO sfinfo;
+        int	readcount;
+
+        // The SF_INFO struct must be initialized before using it
+        memset (&sfinfo, 0, sizeof (sfinfo));
+        if (!(infile = sf_open (audioFilePath[cnt].c_str(), SFM_READ, &sfinfo)))
+        {
+            sf_close (infile);
+            continue;
+        }
+
+        maxSrcWidth = std::max(maxSrcWidth, static_cast<int>(sfinfo.frames));
+        maxChannels = std::max(maxChannels, static_cast<int>(sfinfo.channels));
+
+        // Close input
+        sf_close (infile);
+    }
+    maxDstWidth = maxSrcWidth;
+
+    // Set numDims, offset, n/c/h/w values for src/dst
+    srcDescPtr->numDims = 4;
+    srcDescPtr->offsetInBytes = 0;
+    srcDescPtr->n = batchSize;
```

Comment:
Can't we use set_descriptor_dims_and_strides() like in Tensor_host.cpp?
Also, don't we have calls to these below?
```
// Set src/dst layout types in tensor descriptors
set_descriptor_layout(srcDescPtr, dstDescPtr, layoutType, pln1OutTypeCase, outputFormatToggle);
// Set src/dst data types in tensor descriptors
set_descriptor_data_type(inputBitDepth, funcName, srcDescPtr, dstDescPtr);
```

Reply:
For audio there is no input layout
So we are not able to use the functions defined for image in audio

Reply:
Add few more helpers to modularize the code